nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/avro/AvroRecordSetWriter.java

```diff
@@ -68,16 +76,40 @@
         .required(true)
         .build();
 
+    static final PropertyDescriptor ENCODER_POOL_SIZE = new Builder()
```

Comment:
Since this configuration is performance related and depends on environment, I'd suggest supporting variable registry EL.